program/src/processor.rs

```diff
@@ -300,7 +306,8 @@ pub fn process_stake_deposit(
     // record that here; we will discover it later in `WithdrawInactiveStake`,
     // and then it will be treated as a donation.
     msg!("Staked {} out of the reserve.", amount);
-    validator.entry.stake_accounts_balance = (validator.entry.stake_accounts_balance + amount)?;
+    validator.stake_accounts_balance = (validator.stake_accounts_balance + amount)?;
+    validator.effective_stake_balance = validator.get_effective_stake_balance();
```

Comment:
Hmm, if there is a field `effective_state_balance` and also a method to get it, they can be confused easily, and there may be a risk of the field going out of sync with the getter. Maybe we can make the method mutate the validator instead? Something like `validator.recompute_effective_stake_balance()`?

Reply:
Hmm, so are there cases where the field and the computed value diverge? Or is the field only an optimization?

Reply:
There are no such places in protocol. But I guess it is in `balance.rs`.